features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/featureflags/FlagsClient.kt

```diff
@@ -10,114 +10,175 @@ import com.datadog.android.Datadog
+         * the default Datadog SDK instance will be used.
+         * @return `true` if the client was successfully registered, `false` if a client was
+         * already registered for this SDK core (in which case a warning will be logged).
+         */
+        internal fun registerIfAbsent(client: FlagsClient, sdkCore: SdkCore = Datadog.getInstance()): Boolean =
+            synchronized(registeredClients) {
+                if (registeredClients.containsKey(sdkCore)) {
+                    (sdkCore as FeatureSdkCore).internalLogger.log(
+                        InternalLogger.Level.WARN,
+                        InternalLogger.Target.USER,
+                        { "A FlagsClient has already been registered for this SDK instance" }
+                    )
+                    false
+                } else {
+                    @Suppress("UnsafeThirdPartyFunctionCall") // User provided callable, let it throw
+                    registeredClients[sdkCore] = client
+                    true
+                }
+            }
 
-    internal fun clear() {
-        synchronized(registeredProviders) {
-            registeredProviders.clear()
+        /**
+         * Unregisters the [FlagsClient] associated with the given SDK core.
+         *
+         * After calling this method, subsequent calls to [instance] for the same SDK core
+         * will return a [NoOpFlagsClient]. This method is thread-safe and will silently
+         * do nothing if no client was registered for the given SDK core.
+         *
+         * @param sdkCore the [SdkCore] instance to unregister the client for. If not provided,
+         * the default Datadog SDK instance will be used.
+         */
+        internal fun unregister(sdkCore: SdkCore = Datadog.getInstance()) {
+            synchronized(registeredClients) {
+                registeredClients.remove(sdkCore)
+            }
         }
-    }
 
-    // This method is mainly for test purposes.
-    @Suppress("unused")
-    @JvmStatic
-    private fun reset() {
-        clear()
-    }
+        /**
+         * Removes all registered [FlagsClient] instances from all SDK cores.
+         *
+         * After calling this method, all subsequent calls to [instance] will return
+         * [NoOpFlagsClient] instances regardless of the SDK core. This method is thread-safe
+         * and is primarily intended for testing purposes or SDK shutdown scenarios.
+         */
+        internal fun clear() {
+            synchronized(registeredClients) {
+                registeredClients.clear()
+            }
+        }
 
-    // endregion
+        /**
+         * Resets the internal state by clearing all registered clients.
+         *
+         * This method is intended for testing purposes only and should not be used
+         * in production code. It delegates to [clear] to remove all client registrations.
+         */
+        @Suppress("unused")
+        private fun reset() {
```

Comment:
is it used anywhere? seems like redundant method if it delegates to another one anyway.

Reply:
done